src/Evolution/Systems/GeneralizedHarmonic/Initialize.hpp

```diff
@@ -0,0 +1,175 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <tuple>
+#include <utility>  // IWYU pragma: keep
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Domain/Mesh.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/Initialization/DiscontinuousGalerkin.hpp"
+#include "Evolution/Initialization/Domain.hpp"
+#include "Evolution/Initialization/Evolution.hpp"
+#include "Evolution/Initialization/Interface.hpp"
+//#include "Evolution/Initialization/Limiter.hpp"
+#include "Evolution/Initialization/NonConservativeSystem.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+//#include "Evolution/VariableFixing/FixToAtmosphere.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.tpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace GeneralizedHarmonic {
+namespace Actions {
+namespace detail {
+// Note: this verifies if the Metavariables struct has an `analytic_solution`
+// alias which we could use to initialize variables here.
+template <class T, class = cpp17::void_t<>>
+struct has_analytic_solution_alias : std::false_type {};
+template <class T>
+struct has_analytic_solution_alias<T,
+                                   cpp17::void_t<typename T::analytic_solution>>
+    : std::true_type {};
+}  // namespace detail
+
+template <size_t Dim>
+struct Initialize {
+  template <typename System>
+  struct GrTags {
+    using gr_tag = typename System::variables_tags;
+    using simple_tags = db::AddSimpleTags<gr_tag>;
+    using compute_tags = db::AddComputeTags<>;
```

Comment:
I've recently realized that you'll need a bunch of compute items here, which weren't yet written. I've written most of them in #1358. For instance, you'll need to compute Christoffel symbols, and you'll need lapse and shift, etc.